src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1967 @@ public static void TestBlockingCollectionBasic()
+
+        /// <summary>Validates the Dispose() method.</summary>
+        /// <returns>True if test succeeded, false otherwise.</returns>
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest4_Dispose()
+        {
+            BlockingCollection<int> blockingCollection = ConstructBlockingCollection<int>();
+
+            blockingCollection.Dispose();
+            bool testSuceeded = false;
+            int numOfExceptionsThrown = 0;
+            int numOfTests = 0;
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.Add(default(int));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int), 1);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryAdd(default(int), new TimeSpan(1));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            int item;
+            try
+            {
+                numOfTests++;
+                blockingCollection.Take();
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item, 1);
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            try
+            {
+                numOfTests++;
+                blockingCollection.TryTake(out item, new TimeSpan(1));
+            }
+            catch (ObjectDisposedException)
+            {
+                numOfExceptionsThrown++;
+            }
+
+            const int NUM_OF_COLLECTIONS = 10;
+            BlockingCollection<int>[] blockingCollections = new BlockingCollection<int>[NUM_OF_COLLECTIONS];
+            for (int i = 0; i < NUM_OF_COLLECTIONS - 1; ++i)
+            {
+                blockingCollections[i] = ConstructBlockingCollection<int>(-1);
+            }
+
+            blockingCollections[NUM_OF_COLLECTIONS - 1] = blockingCollection;
```

Comment:
I didn't look into every test cases. I've fixed some very obvious issues, such as replace the wait task in for-loop with WaitAll. Almost all tests (not just this library) are adopted from ancient tests written years ago for desktop. One of the reasons is to make sure the CoreCLR behaviors same as Desktop as much as possible; it also speeds up testing.
If we took this opportunity to clean up all tests, I'm all for it. I'll take a look at these tests.